src/metabase/driver/query_processor.clj

```diff
@@ -107,6 +111,15 @@
            (fail query e)))))
 
 
+(defn- pre-add-settings [qp]
+  (fn [{:keys [driver] :as query}]
+    (let [settings (->> {:report-timezone (when (driver/driver-supports? driver :set-timezone)
+                                            (setting/get :report-timezone))}
+                        (filter (comp not nil? val))
+                        (into {}))]
+      (qp (assoc query :settings settings)))))
```

Comment:
I think this would be more readable written this way
```clojure
(into {} (for [[k v] {...}
               :when (not (nil? v))]
           {k v})
```